### PR TITLE
Port Notebook PRs patching tornado 6

### PR DESCRIPTION
**Pull request by Zsailer**
Porting two PRs:

* [x] jupyter/notebook#4449
* [x] jupyter/notebook#4453

### CODE REVIEW

File: jupyter_server/files/handlers.py [hunk 1]
```diff
@@ -5,12 +5,15 @@
 
 import mimetypes
 import json
+from base64 import decodebytes
```
Comment (Zsailer) [inline jupyter_server/files/handlers.py]: Duplicate

File: jupyter_server/services/kernels/handlers.py [hunk 2]
```diff
@@ -45,8 +45,8 @@ def post(self):
         else:
             model.setdefault('name', km.default_kernel_name)
 
-        kernel_id = yield gen.maybe_future(km.start_kernel(kernel_name=model['name']))
-        model = km.kernel_model(kernel_id)
+        kernel_id = yield maybe_future(km.start_kernel(kernel_name=model['name']))
+        model = yield maybe_future(km.kernel_model(kernel_id))
```
Comment (Zsailer) [inline jupyter_server/services/kernels/handlers.py]: Flagging to double check this change...

File: jupyter_server/services/kernels/handlers.py [hunk 3]
```diff
@@ -83,12 +83,12 @@ def post(self, kernel_id, action):
...
 
             try:
-                yield gen.maybe_future(km.restart_kernel(kernel_id))
+                yield maybe_future(km.restart_kernel(kernel_id))
             except Exception as e:
                 self.log.error("Exception restarting kernel", exc_info=True)
                 self.set_status(500)
             else:
-                model = km.kernel_model(kernel_id)
+                model = yield maybe_future(km.kernel_model(kernel_id))
```
Comment (Zsailer) [inline jupyter_server/services/kernels/handlers.py]: flagging to double check.

File: setup.py [hunk 4]
```diff
@@ -74,7 +74,7 @@
     zip_safe = False,
     install_requires = [
         'jinja2',
-        'tornado>=4, <6',
+        'tornado>=5.0',
```
Comment (kevin-bates) [inline setup.py]: :+1:

**Comment by Zsailer** [comment]
Fixing tests. Looks like some things failed the cherry-pick

**Comment by kevin-bates** [comment]
No cleanup necessary.

**Pull request merged**